File: mobile/MultiTenantEcommerce.Maui/Models/ThemeModels.cs
using System;
using System.Collections.Generic;

namespace MultiTenantEcommerce.Maui.Models;

public record ThemeVariable(string Key, string Value);

public record ThemeSection(string Id, string SectionName, string JsonConfig, int SortOrder);

public record ThemeSummary(
    string Id,
    string Name,
    string Code,
    string Version,
    string? Description,
    string? PreviewImageUrl,
    bool IsActive,
    DateTime CreatedAt,
    IReadOnlyList<ThemeSection> Sections);

public record TenantTheme(
    string TenantThemeId,
    string TenantId,
    ThemeSummary Theme,
    DateTime ActivatedAt,
    bool IsActive,
    IReadOnlyList<ThemeVariable> Variables);

File: frontend/angular/src/app/core/models/theme.models.ts
export interface ThemeSection {
  id: string;
  sectionName: string;
  jsonConfig: string;
  sortOrder: number;
}

export interface ThemeSummary {
  id: string;
  name: string;
  code: string;
  version: string;
  description?: string;
  previewImageUrl?: string;
  isActive: boolean;
  createdAt: string;
  sections: ThemeSection[];
}

export interface ThemeVariable {
  key: string;
  value: string;
}

export interface TenantTheme {
  tenantThemeId: string;
  tenantId: string;
  theme: ThemeSummary;
  activatedAt: string;
  isActive: boolean;
  variables: ThemeVariable[];
}

export interface ThemePreviewResponse {
  previewUrl: string;
  expiresAt: string;
}

export interface ThemeCloneRequest {
  sourceTenantId: string;
  targetTenantId: string;
}

export interface ThemeCloneResponse {
  tenantThemeId: string;
}

export interface ThemeAnalyticsSummary {
  themeId: string;
  themeName: string;
  activeTenantsCount: number;
  averageActiveDays: number;
  topTenants: TenantUsageSnapshot[];
}

export interface TenantUsageSnapshot {
  tenantId: string;
  totalActiveDays: number;
  activatedAt?: string | null;
}

export interface TenantThemeUsage {
  tenantId: string;
  themeId: string;
  isActive: boolean;
  activatedAt: string;
  deactivatedAt?: string | null;
  totalActiveDays: number;
}

export interface ThemeSectionsRequest {
  sections: Array<{
    sectionName: string;
    configuration: unknown;
    sortOrder: number;
  }>;
}

export interface ThemeVariablesRequest {
  variables: Record<string, string>;
}
